### PR TITLE
RFC: Block Middleware

### PR DESCRIPTION
**Pull request by defunctl**
## What does this do/fix?

- Adds a Block Middleware system.
- Adds a global color theme system, that can be applied to blocks or inside block fields.
- Replaces the Tribe ACF Post List Field with a Post Loop Field via Block Middleware.
- Replaces the `Colors` class with a `Swatch_Collection` from square1-field-models.

---
I highly suggest you watch the screencasts before reviewing, pasted from the README:

### Video Guides

- [Block Middleware Overview](https://d.pr/K963cu)

#### Color Theme Middleware

- [How to add the Color Theme Middleware to an existing block](https://d.pr/aIi5HY)
- ~~[How to add the Color Theme Middleware to a repeater](https://d.pr/cUAx8b)~~ - outdated.

### Post Loop Field Middleware

- [Post Loop Field Overview](https://d.pr/Px5xSe)
- ~[How to configure the Post Loop Field Middleware](https://d.pr/FicVgp)~ outdated
- [How to use the Post Loop Field in non-block controllers](https://d.pr/UP7fcL)

---
## TODO
- [x] Remove composer Tribe Libs branch version once https://github.com/moderntribe/tribe-libs/pull/126 is released. 

---

## QA

See screencasts above.

## Tests

Does this have tests?

- [x] Yes
- [ ] No, this doesn't need tests because...
- [ ] No, I need help figuring out how to write the tests.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -100,9 +100,10 @@
     "moderntribe/acf-menu-chooser": "^1.1",
     "moderntribe/square1-field-models": "^1.0",
     "moderntribe/tribe-acf-post-list-field": "^2.0",
-    "moderntribe/tribe-libs": "^3.4",
+    "moderntribe/tribe-libs": "dev-feature/block-middleware as 3.4.18",
```
Comment (defunctl) [inline composer.json:103]: Note: Revert this once the companion PR is released: https://github.com/moderntribe/tribe-libs/pull/123

**Comment by tarecord** [comment]
That overview video is awesome 🌮 A lot going on, but good explanation!

File: wp-content/plugins/core/src/Blocks/Middleware/Post_Loop/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# Post Loop Middleware
```
Comment (ChrisMKindred) [inline wp-content/plugins/core/src/Blocks/Middleware/Post_Loop/README.md:1]: Is there more you were wanting to add here?

**Review by ChrisMKindred — APPROVED** [review]
This is great @defunctl, Super powerful but also not something that must be used unless there is you need it and then it is there.

File: wp-content/plugins/core/src/Blocks/Middleware/Post_Loop/README.md (hunk 2 above)
Comment (defunctl) [reply wp-content/plugins/core/src/Blocks/Middleware/Post_Loop/README.md]: Removed this file.

**Pull request merged**